src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1041,6 +1114,59 @@ internal override bool HasBackingField(SyntaxNode propertyOrIndexerDeclaration)
         internal override bool IsDeclarationWithInitializer(SyntaxNode declaration)
             => declaration is VariableDeclaratorSyntax { Initializer: not null } || declaration is PropertyDeclarationSyntax { Initializer: not null };
 
+        internal override bool IsRecordPrimaryConstructorParameter(SyntaxNode declaration)
+            => declaration is ParameterSyntax { Parent: ParameterListSyntax { Parent: RecordDeclarationSyntax } };
+
+        internal override bool IsPropertyDeclarationMatchingPrimaryConstructorParameter(SyntaxNode declaration)
+            => declaration is PropertyDeclarationSyntax { Parent: RecordDeclarationSyntax record, Identifier: { ValueText: var name } }
+            && record.ParameterList is not null
+            && record.ParameterList.Parameters.Any(p => p.Identifier.ValueText.Equals(name));
+
+        internal override bool IsPropertyAccessorDeclarationMatchingPrimaryConstructorParameter(SyntaxNode declaration, out bool isFirstAccessor)
+        {
+            isFirstAccessor = false;
+            if (declaration is AccessorDeclarationSyntax { Parent: AccessorListSyntax { Parent: PropertyDeclarationSyntax property } list } &&
+                IsPropertyDeclarationMatchingPrimaryConstructorParameter(property))
+            {
+                isFirstAccessor = list.Accessors[0] == declaration;
+                return true;
+            }
+            return false;
+        }
+
+        internal override IEnumerable<ISymbol> GetRecordUpdatedSynthesizedMembers(Compilation compilation, INamedTypeSymbol record)
```

Comment:
>GetRecordUpdatedSynthesizedMembers [](start = 47, length = 34)

Since this operates on symbols, does it need to be virtual with an implementation in C#?

Reply:
Nope, will move